Webapp/sources/services/solvent_surfer/utils.py

```diff
@@ -472,8 +472,10 @@ def jsonify_embedding_params(embedding_algorithm: cPCA) -> json.dumps:
 
     embedding_algorithm_dict = embedding_algorithm.__dict__
 
+    save_dict = embedding_algorithm_dict.copy()
+
     # remove python classes from dict (these can't be converted to JSON)
-    del embedding_algorithm_dict['parent']
-    del embedding_algorithm_dict['embedder']
+    del save_dict['parent']
+    del save_dict['embedder']
 
-    return json.dumps(embedding_algorithm_dict, cls=NumpyEncoder)
+    return json.dumps(save_dict, cls=NumpyEncoder)
```

Comment:
a more pythonic way of this might be to exclude the keys upfront:

```
excluded_keys = ['parent', 'embedder']
save_dict = {key: value for key, value in embedding_algorithm.__dict__.items() if key not in excluded_keys}
    return json.dumps(save_dict, cls=NumpyEncoder)
```

also the return type of the function is just a string - as that's what `json.dumps` will return.

Reply:
Thanks, I'll address this in the next pull request just to save you having to re-review this one